### PR TITLE
Release funasr-onnx 0.4.2 without Torch runtime dependency

### LINKED ISSUE #2604: funasr_onnx移除对torch依赖
funasr_onnx已经实现了基于onnx runtime。
但是在sensevoice_bin.py代码的__call__函数中，两处地方依赖了torch库。
- torch.from_numpy(ctc_logits)
- torch.unique_consecutive(yseq, dim=-1)

这两个函数完全可以通过其他方式支持，从而避免依赖torch库（几百M）

### PR DESCRIPTION
**Pull request by LauraGPT**
## Summary

- bump `funasr-onnx` from `0.4.1` to the immutable `0.4.2` release candidate;
- add a release-contract test that requires `jieba`, rejects Torch as a runtime dependency, and rejects Torch imports in the ONNX package source;
- add Python 3.11/3.12 CI that builds the wheel and sdist, runs `twine check`, installs the wheel in a clean environment, runs `pip check`, imports all four public entry points, and uploads the release candidate artifact.

## Why

The source fixes for #2604 are already on `main`: SenseVoice post-processing is NumPy-only and the missing `jieba` dependency is declared. PyPI still serves `funasr-onnx==0.4.1`, however, so users continue to receive the old Torch-importing wheel. Re-uploading the existing version is not safe; this PR creates a new version and makes the release contract reproducible.

This PR does **not** close #2604 by itself. Keep the issue open until `0.4.2` is uploaded to PyPI and the downloaded PyPI wheel passes the same smoke test.

## Verification

- release contract: 4 tests passed;
- `python -m build`: wheel and sdist built successfully;
- `twine check`: both distributions passed;
- fresh Python 3.11.15 and 3.12.3 environments: 34 packages compatible, `funasr-onnx==0.4.2`, Torch absent, `jieba` present, and `Paraformer`, `Fsmn_vad`, `CT_Transformer`, `SenseVoiceSmall` all import successfully;
- real CPU inference from the newly built wheel, with Torch absent, using the official quantized SenseVoice ONNX model:
  - input: 16 kHz mono, 5.55 seconds;
  - output: `<|zh|><|NEUTRAL|><|Speech|><|withitn|>欢迎大家来体验达摩院推出的语音识别模型。`;
- Black, compileall, workflow semantic checks, and `git diff --check` passed.

Local release-candidate checksums:

```text
42243d2d5f91b679220206bdc4424861127bb7814693fae38ce4026f4e10cbca  funasr_onnx-0.4.2-py3-none-any.whl
c13e1d6b832c902afc6716a07d9ed70ec33394102a34bb8c0f3bb87ccfe65435  funasr_onnx-0.4.2.tar.gz
```

## Post-merge release gate

A maintainer still needs to upload the final artifacts with the project PyPI credential. After upload, download `funasr-onnx==0.4.2` from PyPI into fresh Python 3.11 and 3.12 environments and repeat the no-Torch dependency/import smoke before closing #2604.

### CODE REVIEW

File: tests/test_funasr_onnx_release.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+            .split("=", 1)[0]
+            .split("<", 1)[0]
+            .split(">", 1)[0]
+            .strip()
+            .lower()
+            .replace("_", "-")
+            for requirement in requirements
+        }
+        self.assertIn("jieba", names)
+        self.assertNotIn("torch", names)
```
[the remark this answers is not in the document]
Comment (LauraGPT) [reply tests/test_funasr_onnx_release.py:54]: Fixed in `ca024b8dd`. The contract now extracts the distribution name with an anchored standard-library regex, normalizes `-_.` consistently, and has regression cases for `!=`, `~=`, extras plus markers, and direct URL requirements. The four contract tests pass on Python 3.11 and 3.12 locally, and both jobs in Actions run 29491518970 are green.
